### PR TITLE
fix: don't append Shift modifier text twice to accelerators (backport: 4-0-x)

### PR DESCRIPTION
**Pull request by brenca**
#### Description of Change

Backport of #15400 to 4-0-x
#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: fixed some accelerators having `Shift` appended to them twice

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
@brenca Can you manually confirm what the `Control+Shift+=` accelerator looks like and what the `Control++` accelerator looks like after this PR?

**Comment by brenca** [comment]
@MarshallOfSound 
Before:
image

After:
image

```
const template = [
    {
      label: "Menu item",
      submenu: [
        {
          label: "One",
          id: "one",
          type: "normal",
          accelerator: "Shift+Backspace",
          click: () => {
            console.log('asd');
          }
        },
        {
          label: "Two",
          id: "two",
          type: "normal",
          accelerator: "Control+Shift+=",
          click: () => {
            console.log('asd');
          }
        },
        {
          label: "Three",
          id: "three",
          type: "normal",
          accelerator: "Control++",
          click: () => {
            console.log('asd');
          }
        }
      ]
    }
  ];
```

**Comment by brenca** [comment]
@MarshallOfSound if I use `Control+Plus` it shows up as `Control+Shift+=` in all versions

**Comment by nornagon** [comment]
Would it be possible to encode @MarshallOfSound's intuition about what might have broken into a test? 🤞

**Comment by brenca** [comment]
@nornagon AFAIK what @MarshallOfSound asked about is how it gets displayed, which is not really testable from JS, but I might be wrong on that.

**Comment by nornagon** [comment]
Could you potentially expose the result of `Accelerator::GetShortcutText` to JS and test it that way?

**Comment by brenca** [comment]
@nornagon Yeah, that could work

File: spec/api-menu-item-spec.js [hunk 1]
```diff
@@ -389,4 +389,64 @@ describe('MenuItems', () => {
...
+
+    it('should display modifiers correctly for special keys', () => {
+      const menu = Menu.buildFromTemplate([
+        { label: 'text', accelerator: 'CmdOrCtrl+Tab' },
+        { label: 'text', accelerator: 'Shift+Tab' },
+        { label: 'text', accelerator: 'Alt+Tab' }
+      ])
+
+      assert.strictEqual(menu.getAcceleratorTextAt(0),
+        isDarwin() ? '⌘⇥\u0000' : 'Ctrl+Tab')
```
Comment (nornagon) [inline spec/api-menu-item-spec.js]: where are these `\u0000` coming from? that doesn't seem right...

Comment (nornagon) [reply spec/api-menu-item-spec.js]: that's where _you_ got them from, but not where they _came_ from :P

File: spec/api-menu-item-spec.js [hunk 2]
```diff
@@ -389,4 +389,68 @@ describe('MenuItems', () => {
...
+      const menu = Menu.buildFromTemplate([
+        { label: 'text', accelerator: 'Control+Shift+=' },
+        { label: 'text', accelerator: 'Control+Plus' }
+      ])
+
+      assert.strictEqual(menu.getAcceleratorTextAt(0),
+        isDarwin() ? '⌃⇧=' : 'Ctrl+Shift+=')
+      assert.strictEqual(menu.getAcceleratorTextAt(1),
+        isDarwin() ? '⌃⇧=' : 'Ctrl+Shift+=')
+      done()
```
[the remark this answers is not in the document]
Comment (alexeykuzmin) [reply spec/api-menu-item-spec.js]: Done in #15550.

File: spec/api-menu-item-spec.js (hunk 1 above)
Comment (brenca) [reply spec/api-menu-item-spec.js]: @nornagon my best guess would be [that 0 at the end](https://cs.chromium.org/chromium/src/ui/base/accelerators/accelerator.cc?rcl=cc75805c1f32cad6601f75cf4cadfbee2344637d&l=296) that we (apparently) keep during conversion from `base::string16`

**Pull request merged**